src/store/index.ts

```diff
@@ -1,51 +1,128 @@
 import { createStore, AnyAction } from 'redux';
 import { composeWithDevTools } from 'redux-devtools-extension';
 
-// Action types - is just a constant. MUST have a unique value.
+const SORT_BY_OPTIONS = {
+  title: 'title',
+  completed: 'completed',
+  userName: 'userName',
+};
+
+const ORDER = {
+  asc: 'asc',
+  desc: 'desc',
+};
+
 const START_LOADING = 'START_LOADING';
+const SET_TODOS = 'SET_TODOS';
+const HANDLE_ERROR = 'HANDLE_ERROR';
 const FINISH_LOADING = 'FINISH_LOADING';
+const DELETE_TODO = 'DELETE_TODO';
+const SET_STATUS = 'SET_STATUS';
+const SORT_BY = 'SORT_BY';
 
-// Action creators - a function returning an action object
 export const startLoading = () => ({ type: START_LOADING });
-export const finishLoading = (message = 'No message') => ({ type: FINISH_LOADING, message });
+export const finishLoading = () => ({ type: FINISH_LOADING });
 
-// Selectors - a function receiving Redux state and returning some data from it
-export const isLoading = (state: RootState) => state.loading;
-export const getMessage = (state: RootState) => state.message;
+export const setTodos = (todos: Todo[]) => ({ type: SET_TODOS, todos });
+export const handleError = (errorMessage: string) => ({ type: HANDLE_ERROR, errorMessage });
+export const deleteTodo = (id: number) => ({ type: DELETE_TODO, id });
+export const setStatus = (id: number) => ({ type: SET_STATUS, id });
+export const sortBy = (field: string) => ({ type: SORT_BY, field });
+
+export const isLoading = (state: RootState) => state.isLoading;
+export const hasError = (state: RootState) => state.errorMessage;
```

Comment:
this is not a boolean value, this is a function => `getHasError`

the same for others